ocf_data_sampler/torch_datasets/sample/uk_regional.py

```diff
@@ -22,22 +24,139 @@ def __init__(self, data: NumpySample) -> None:
 
     @override
     def to_numpy(self) -> NumpySample:
+        """Returns the data as a NumPy sample."""
         return self._data
 
     @override
     def save(self, path: str) -> None:
+        """Saves sample to the specified path in pickle format."""
         # Saves to pickle format
         torch.save(self._data, path)
 
     @classmethod
     @override
     def load(cls, path: str) -> "UKRegionalSample":
+        """Loads sample from the specified path.
+
+        Args:
+            path: Path to the saved sample file.
+
+        Returns:
+            A UKRegionalSample instance with the loaded data.
+        """
         # Loads from .pt format
         # TODO: We should move away from using torch.load(..., weights_only=False)
         return cls(torch.load(path, weights_only=False))
 
+    def validate_sample(self, config: dict | None = None) -> dict:
+        """Validates that the sample has the expected structure and data shapes.
+
+        Args:
+            config: Optional configuration dict with expected shapes and required fields.
+                If None, uses default validation rules.
+
+        Returns:
+            dict: Validation results with status and any validation errors.
+        """
+        validation_result = {
+            "valid": True,
+            "errors": [],
+        }
```

Comment:
I can see why you have done this, im be tempted to first of all keep it simple, and just assert errors